### PR TITLE
Add SocketsHttpHandler requests-queue-duration metrics

### PR DESCRIPTION
**Pull request by MihaZupan**
Contributes to #88384

Following #88893, this PR adds the `http-client-requests-queue-duration` counter (final name TBD).
Includes the same tags as other connection counters: `protocol`, `scheme`, `host`, and `port`.

### CODE REVIEW

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Metrics/SocketsHttpHandlerMetrics.cs [hunk 1]
```diff
@@ -19,5 +20,42 @@ internal sealed class SocketsHttpHandlerMetrics(Meter meter)
...
+            if (RequestsQueueDuration.Enabled)
+            {
+                TagList tags = default;
+
+                tags.Add("protocol", versionMajor switch
+                {
+                    1 => "HTTP/1.1",
+                    2 => "HTTP/2",
+                    _ => "HTTP/3"
+                });
```
Comment (antonfirsov) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Metrics/SocketsHttpHandlerMetrics.cs:42]: I assume it makes more sense to conform to other connection metrics and existing event counters, but note that request metrics log `HTTP/1.0` if `HttpRequestMessage.Version` is `1.0`. Is it worth a comment?

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs [hunk 2]
```diff
@@ -998,19 +998,20 @@ private async ValueTask<Http3Connection> GetHttp3ConnectionAsync(HttpRequestMess
                     ThrowGetVersionException(request, 3, reasonException);
                 }
 
-                long queueStartingTimestamp = HttpTelemetry.Log.IsEnabled() ? Stopwatch.GetTimestamp() : 0;
+                bool requestsQueueDurationEnabled = HttpTelemetry.Log.IsEnabled() || Settings._metrics!.RequestsQueueDuration.Enabled;
+                long queueStartingTimestamp = requestsQueueDurationEnabled ? Stopwatch.GetTimestamp() : 0;
```
Comment (antonfirsov) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs:1002]: Would it make things simpler if we had a single `long? queueStartingTimestamp` parameter here?

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs [hunk 3]
```diff
@@ -998,19 +998,20 @@ private async ValueTask<Http3Connection> GetHttp3ConnectionAsync(HttpRequestMess
...
 
                 ValueTask<Http3Connection> connectionTask = GetHttp3ConnectionAsync(request, authority, cancellationToken);
 
-                if (HttpTelemetry.Log.IsEnabled() && connectionTask.IsCompleted)
+                if (requestsQueueDurationEnabled && connectionTask.IsCompleted)
                 {
                     // We avoid logging RequestLeftQueue if a stream was available immediately (synchronously)
```
Comment (antonfirsov) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs:1008]: I'm confused by this comment. It states that *We avoid logging RequestLeftQueue*, but then `Http3Connection.SendAsync` requests a new timestamp before `OpenOutboundStreamAsync` and actually logs it.

(And is it actually good that `OpenOutboundStreamAsync` is part of the "request queue duration" measurement?)

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Metrics/SocketsHttpHandlerMetrics.cs [hunk 4]
```diff
@@ -19,5 +20,42 @@ internal sealed class SocketsHttpHandlerMetrics(Meter meter)
...
+                    tags.Add("port", pool.OriginAuthority.Port);
+                }
+
+                RequestsQueueDuration.Record(duration.TotalSeconds, tags);
+            }
+
+            if (HttpTelemetry.Log.IsEnabled())
+            {
+                HttpTelemetry.Log.RequestLeftQueue(versionMajor, duration);
+            }
```
Comment (antonfirsov) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Metrics/SocketsHttpHandlerMetrics.cs:58]: Nit: maybe it's ok do dupe this out to callers so we don't do `HttpTelemetry` implicitly in metrics methods?

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Metrics/SocketsHttpHandlerMetrics.cs (hunk 1 above)
Comment (MihaZupan) [reply src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Metrics/SocketsHttpHandlerMetrics.cs:42]: I think it makes sense for requests to log 1.0, but for connections, we only really have HTTP/1.1 that we share between 1.0 and 1.1 requests.
Added a small comment.

**Review by stephentoub — APPROVED** [review]

**Review by antonfirsov — APPROVED** [review]

**Comment by MihaZupan** [comment]
Test failure is #87477

**Pull request merged**